File: src/main/java/pr/eleks/we_at_her/repositories/WeatherSampleRepository.java
package pr.eleks.we_at_her.repositories;

import org.springframework.data.repository.CrudRepository;
import pr.eleks.we_at_her.entities.WeatherSample;

import java.util.Optional;

public interface WeatherSampleRepository extends CrudRepository<WeatherSample, Long> {
    Optional<WeatherSample> findFirstByCityIdAndTime(int cityId, int time);
}
// TODO fix this returning null

File: src/main/kotlin/pr/eleks/we_at_her/repositories/WeatherSampleRepository.kt
package pr.eleks.we_at_her.repositories

import org.springframework.data.repository.CrudRepository
import pr.eleks.we_at_her.entities.WeatherSample
import java.util.*

interface WeatherSampleRepository : CrudRepository<WeatherSample, Long> {
    fun findFirstByCityIdAndTime(cityId: Int, time: Int): WeatherSample?
}